extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ConfigStaticInitCheckInterceptor.java

```diff
@@ -37,7 +36,8 @@ public class ConfigStaticInitCheckInterceptor {
 
     @AroundInvoke
     Object aroundInvoke(InvocationContext context) throws Exception {
-        recordConfigValue(null, configValues);
+        InjectionPoint injectionPoint = (InjectionPoint) context.getParameters()[0];
```

Comment:
Ech, this looks very fragile, what if the signature of those producer methods is changed?

Reply:
Yeah, so this was the only place where there's no `CreationalContext`. Fortunately, all the methods obtain the `InjectionPoint` as their first (and only) parameter, so it's alright, we can just lookup the parameter from the `InvocationContext`. If we cannot do that, we cannot remove the thread-local, which would be unfortunate.

This interceptor only intercepts the `@Produces @ConfigProperty` methods  on `io.smallrye.config.inject.ConfigProducer`, which looks like a fairly niche use case and shouldn't really affect anyone IMHO.

Reply:
I see. In that case, we should make the code a little bit more robust because quarkus does not control the `io.smallrye.config.inject.ConfigProducer`, i.e. at least iterate over the method arguments and match the correct one by type... WDYT?

Reply:
I thought about that and figured it's not necessary, because there are no other arguments. But you're right it will make the code a bit more future-proof, so I can do that.

Reply:
Done.